src/Entity/EntityIdValue.php

```diff
@@ -121,10 +129,20 @@ public function getArrayValue() {
 		return array(
 			'entity-type' => $this->entityId->getEntityType(),
 			'numeric-id' => $this->getNumericId(),
+			'id' => $this->entityId->getSerialization(),
 		);
 	}
 
 	/**
+	 * @since 6.1
+	 *
+	 * @param EntityIdParser $idParser
+	 */
+	public static function setEntityIdParser( EntityIdParser $idParser ) {
```

Comment:
Yea... probably the best we can do without overhauling the entire infrastructure. But we *should* overhaul the entire infrastructure. This sucks.